borgmatic/hooks/pushover.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def initialize_monitor(
+    ping_url, config, config_filename, monitoring_log_level, dry_run
+):  # pragma: no cover
+    '''
+    No initialization is necessary for this monitor.
+    '''
+    pass
+
+
+def ping_monitor(hook_config, config, config_filename, state, monitoring_log_level, dry_run):
+    '''
+    Post a message to the configured Pushover application.
+    If this is a dry run, then don't actually update anything.
+    '''
+
+    run_states = hook_config.get('states', ['fail'])
+
+    if state.name.lower() not in run_states:
+        return
+
+    dry_run_label = ' (dry run; not actually updating)' if dry_run else ''
+
+    state_config = hook_config.get(
+        state.name.lower(),
+        {
+            'message': state.name.lower(),
+        },
+    )
+
+    token = hook_config.get('token')
+    user = hook_config.get('user')
+
+    logger.info(f'{config_filename}: Updating Pushover {dry_run_label}')
+
+    if token is None:
+        logger.warning(f'{config_filename}: Token missing for Pushover')
+        return
+    if user is None:
+        logger.warning(f'{config_filename}: User missing for Pushover')
+        return
+
+    if 'priority' in state_config and state_config['priority'] == 2:
+        if 'expire' not in state_config:
+            logger.info(f'{config_filename}: Setting expire to default (10min).')
+            state_config['expire'] = 1200
+        if 'retry' not in state_config:
+            logger.info(f'{config_filename}: Setting retry to default (30sec).')
+            state_config['retry'] = 30
+    else:
+        state_config.pop('expire', None)
+        state_config.pop('retry', None)
+
+    data = {
+        'token': token,
+        'user': user,
+        'message': state.name.lower(),  # default to state name. Can be overwritten in state_config loop below.
+    }
+    
+    for key in state_config:
+        data[key] = state_config[key]
```

Comment:
A more succinct way to accomplish this would be something like:

```python
data = dict(
    {
        'token': token,
        'user': user,
        'message': state.name.lower(),  # default to state name. Can be overwritten in state_config loop below.
    },
    **state_config,
)
```

Or maybe this is more your style:

```python
data = {
    'token': token,
    'user': user,
    'message': state.name.lower(),  # default to state name. Can be overwritten in state_config loop below.
}

data.update(state_config)

Reply:
that is very cool. I didn't know that was possible: https://github.com/tony1661/borgmatic/commit/b1af304125ff07bf65ac7eaa71f59e2cfff8a25f

Tests all pass with it so I will resolve